src/Core/src/Hosting/HandlerMauiAppBuilderExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static MauiAppBuilder ConfigureMauiHandlers(this MauiAppBuilder builder,
 
 		public static IServiceCollection ConfigureMauiHandlers(this IServiceCollection services, Action<IMauiHandlersCollection>? configureDelegate)
 		{
-			services.TryAddSingleton<IMauiHandlersFactory, MauiHandlersFactory>();
+			services.TryAddSingleton<IMauiHandlersFactory>(sp => new MauiHandlersFactory(sp.GetServices<HandlerRegistration>()));
```

Comment:
Is there a way we make it where we get a build error if we tried call `TryAddSingleton<IMauiHandlersFactory, MauiHandlersFactory>();` internally in dotnet/maui?

But it would be fine if developers want to use it in their apps?

Reply:
One idea that pops to mind is to use the BannedApiAnalyzers.

https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md

Do you think I should hook that up here?

Reply:
If it's easy to set up that might be a good idea... It seems like there are probably other APIs we would put in that text file?

Reply:
Done. It wasn't too bad, and now we can use this functionality in other places as well.